qcodes/dataset/linked_datasets/links.py

```diff
@@ -0,0 +1,84 @@
+"""
+This module defines the Link dataclass as well as two functions to read and
+write an Link object to/from string, respectively
+"""
+from typing import List
+from dataclasses import dataclass, asdict
+import re
+import json
+
+_guid_pattern = re.compile(r'^[\da-f]{8}-([\da-f]{4}-){3}[\da-f]{12}$')
+
+
+@dataclass
+class Link:
+    """
+    Class to represent a link between two datasets. The link is a little graph
+    with two nodes (head and tail) and a directed edge.
+
+    Attributes:
+        head: a guid representing the head of the graph
+        tail: a guid representing the tail of the graph
+        edge_type: a name to represent the type of the edge
+        description: free-form optional field add a description of the graph
+    """
+    head: str
+    tail: str
+    edge_type: str
+    description: str = ""
+
+    @staticmethod
+    def validate_node(node_guid: str, node: str) -> None:
+        """
+        Validate that the guid given is a valid guid.
+
+        Args:
+            node_guid: the guid
+            node: either "head" or "tail"
+        """
+        if _guid_pattern.match(node_guid):
```

Comment:
Should we not refactor this out into function in the guids.py (is_valid_guid or some such thing)

Reply:
Yes, let's do that. Note that the verification is only of the format. Nonsensical values can still slip through.

Reply:
Yes but that can be improved in a future PR